frontend/ui/widget/inputdialog.lua

```diff
@@ -114,28 +152,159 @@ function InputDialog:init()
             width = self.width,
         }
     }
-
+    self.title_bar = LineWidget:new{
+        dimen = Geom:new{
+            w = self.width,
+            h = Size.line.thick,
+        }
+    }
     if self.description then
-        self.description = FrameContainer:new{
-            padding = self.title_padding,
-            margin = self.title_margin,
+        self.description_widget = FrameContainer:new{
+            padding = self.desc_padding,
+            margin = self.desc_margin,
             bordersize = 0,
             TextBoxWidget:new{
                 text = self.description,
                 face = self.description_face,
-                width = self.width - 2*self.title_padding - 2*self.title_margin,
+                width = self.width - 2*self.desc_padding - 2*self.desc_margin,
             }
         }
     else
-        self.description = VerticalSpan:new{ width = self.title_margin + self.title_padding }
+        self.description_widget = VerticalSpan:new{ width = 0 }
+    end
+
+    -- Vertical spaces added before and after InputText
+    -- (these will be adjusted later to center the input text if needed)
+    local vspan_before_input_text = VerticalSpan:new{ width = 0 }
+    local vspan_after_input_text = VerticalSpan:new{ width = 0 }
+    -- We add the same vertical space used under description after the input widget
+    -- (can be disabled by setting condensed=true)
+    if not self.condensed then
+        local desc_pad_height = self.desc_margin + self.desc_padding
+        if self.description then
+            vspan_before_input_text.width = 0 -- already provided by description_widget
+            vspan_after_input_text.width = desc_pad_height
+        else
+            vspan_before_input_text.width = desc_pad_height
+            vspan_after_input_text.width = desc_pad_height
+        end
     end
 
+    -- Buttons
+    if self.add_nav_bar then
+        if not self.buttons then
+            self.buttons = {}
+        end
+        local nav_bar = {}
+        table.insert(self.buttons, nav_bar)
+        table.insert(nav_bar, {
+            text = "⇱",
```

Comment:
Is there a rationale behind this specific set of symbols?

Reply:
No rationale, just did what I could with the fact that
- some arrows/symbols are not available in our fonts
- some others are too thin
- some arrows have only the UP symbol in unicode, and no DOWN - they needed to be symetrical
- the scroll up & down are homogenous with the ones used in dict widget
- these home & end are indeed looking very different, but they somehow indicate quite well what they do

But if while testing you find some other combinations that look fine, I have no undying affection for these ones :)

Reply:
For reference:
http://xahlee.info/comp/unicode_computing_symbols.html
http://xahlee.info/comp/unicode_arrows.html
https://unicode-table.com/en/sets/arrows-symbols/
https://www.key-shortcut.com/en/writing-systems/35-symbols/arrows/